SocketPerfTest/Program.cs

```diff
@@ -115,19 +121,38 @@ static void ShowResults(BaseClientOptions options, ClientHandler[] clientHandler
 
             int elapsed = 0;
             int previousCount = countAfterWarmup;
+            int ReportingInterval = options.ReportingInterval > 0 ? options.ReportingInterval : 1;
```

Comment:
If options.ReportingInterval == 0, why sample every second?  Seems like we could just set reportingInterval = DurationTime and sample once at the end.

Also, we should probably check for the case where ReportingInterval == 0 and DurationTime == 0, and exit with an error; otherwise if someone tries this they will have a bad time.

Reply:
That would make sense. What would you do if we are terminating by number of requests?
With that you can simply fire N number of requests and end without having to worry about time. I know you can keep the reporting interval positive but I was trying to find a way how to suppress periodic updates. Alternatively I can can modify the output and have the final line slightly different. With that I can easily find it if this is called from another script.

Reply:
Yeah, for # of requests termination, what you're doing seems fine (i.e. sample every second).

Reply:
And just to be clear, I think it's fine to be able to disable the reporting interval.  I just want to unsure that the user sets either a limit (time or requests) or a reporting interval, otherwise the app is just going to basically hang and report nothing.